src/deluge/gui/context_menu/save_song_or_instrument.cpp

```diff
@@ -49,12 +49,11 @@ bool SaveSongOrInstrument::acceptCurrentOption() {
 
 	case 1: { // Create folder
 		Browser* browser = (Browser*)getUIUpOneLevel();
-		Error error = browser->createFolder();
-
-		if (error != Error::NONE) {
+		Error error;
+		D_TRY_CATCH(browser->createFolder(), {
```

Comment:
This looks is a little strange now for two reasons:
- The macro just expects Error error to exist in scope (which could create some fun compilation errors if you don't know about that and are using it for the first time)
- If you don't know that the macro uses / assigns error, it would appear reading this that error is never assigned.

It's a subtle change, but I'd recommend changing the macro to accept a name and have it declare the Error internally like:
```
#define D_TRY_CATCH(expr, error, block)    \
	Error error = (expr);                    \
	if (error != Error::NONE) {                 \
		block                                   \
	}
```

Usage:
```
D_TRY_CATCH(browser->createFolder(), error, {
  display->displayError(error);
  close();
  return false;
});
```

Added bonus: one line of code fewer per usage.

Reply:
Also if you do this, I'd probably wrap it in a `do { } while(0};` setup to scope it so that you can safely reuse the same error name when using the macro multiple times in the same scope.